src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -122,6 +122,9 @@ public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDep
                 MethodDesc impl = defType.FindVirtualFunctionTargetMethodOnObjectType(decl);
                 if (impl.OwningType == defType && !impl.IsAbstract)
                 {
+                    if (decl.HasInstantiation)
```

Comment:
We can do this before `FindVirtualFunctionTargetMethodOnObjectType(decl)` to save some cycles.